### PR TITLE
update register and interceptor

### PR DESCRIPTION
**Pull request by Cwagne17**
I updated the interceptor to get the token if it doesn't have it already. This was an issue I found when registering it wouldn't have the token because the provider was already intialized.

Also it updates the body sent to the register route so that the name is split into firstname and lastname.

### CODE REVIEW

File: src/app/core/auth-module/auth.service.ts [hunk 1]
```diff
@@ -388,7 +388,14 @@ export class AuthService {
    *
    * @param user the user's data from the registration component
    */
-  async register(user: any): Promise<User> {
+  async register(user: {
```
Comment (dsoto18) [inline src/app/core/auth-module/auth.service.ts:391]: Looks good, there isn't a User DTO we could use? Just a thought

**Review by johngonz0 — APPROVED** [review]
nice

File: src/app/core/auth-module/auth.service.ts (hunk 1 above)
Comment (Greek) [reply src/app/core/auth-module/auth.service.ts:391]: ^ there's a million places in the client that need proper typing too, would be a good story to complete later on =)

**Review by Greek — APPROVED** [review]

File: src/app/core/auth-module/auth.service.ts (hunk 1 above)
Comment (pzalep1) [reply src/app/core/auth-module/auth.service.ts:391]: Believe it or not, but there used to be an npm package that provided types for everything. It was removed in 2019.

**Review by pzalep1 — APPROVED** [review]

File: src/app/core/auth-module/auth.service.ts (hunk 1 above)
Comment (Cwagne17) [reply src/app/core/auth-module/auth.service.ts:391]: @dsoto18 There is a type in the backend, but like Andreas said, as you'll come to learn there are many refactors needed on the frontend for proper typing.

I do agree we should get to this though

**Pull request merged**